api/metadata/module.go

```diff
@@ -72,13 +72,13 @@ func (m Module) Metadata() int64 {
 
 // TODO: logic is very similar to MetadataAtVersion (for v14). Should be refactored at some point
 func (m Module) buildMetadata() primitives.Metadata {
-	metadataTypesIds := primitives.BuildMetadataTypesIdsMap()
+	metadataGenerator := primitives.NewMetadataTypeGenerator()
```

Comment:
there is another instance on line 111. this should be added as a struct field so it can be instantiated once in the module constructor and then reused within the module methods